src/vanity_key/mod.rs

```diff
@@ -1,80 +1,77 @@
-    bytes_buffer_public.put(&bytes_public);
-    bytes_buffer_public.put_u16::<LittleEndian>(checksum_public);
-    // Base 32 encode the public key
-    let public_key = encode(RFC4648 { padding: false }, &bytes_buffer_public);
+    fn generate_random_key(&mut self) -> (String, String) {
+        // Generate ED25519 key pair
+        let keypair: Keypair = Keypair::generate(&mut self.rng);
 
-    // ************** Encode the private key ***************** //
-    const VERSION_BYTE_SEED: u8 = 18 << 3;
-    let mut bytes_private = vec![VERSION_BYTE_SEED];
-    // Combine the byte version and the ED25519 raw private key bytes array
-    &bytes_private.extend_from_slice(&keypair.secret.to_bytes());
-    // Calculate checksum
-    let check_sum_private = State::<XMODEM>::calculate(&bytes_private);
-    // Create a buffer to combine byte version : ED25519 raw key : checksum
-    let mut bytes_buffer_private = BytesMut::with_capacity(1024);
-    bytes_buffer_private.put(&bytes_private);
-    bytes_buffer_private.put_u16::<LittleEndian>(check_sum_private);
-    // Base 32 encode the private key
-    let private_key = encode(RFC4648 { padding: false }, &bytes_buffer_private);
+        // ************** Encode the public key ***************** //
+        const VERSION_BYTE_ACCOUNT_ID: u8 = 6 << 3;
+        let mut bytes_public = vec![VERSION_BYTE_ACCOUNT_ID];
+        // Combine the byte version and the ED25519 raw public key bytes array
+        &bytes_public.extend_from_slice(&keypair.public.to_bytes());
+        // Calculate checksum
+        let checksum_public = State::<XMODEM>::calculate(&bytes_public);
+        // Create a buffer to combine byte version : ED25519 raw key : checksum
+        let mut bytes_buffer_public = BytesMut::with_capacity(1024);
+        bytes_buffer_public.put(&bytes_public);
+        bytes_buffer_public.put_u16_le(checksum_public);
+        // Base 32 encode the public key
+        let public_key = encode(RFC4648 { padding: false }, &bytes_buffer_public);
 
-    (public_key, private_key)
+        // ************** Encode the private key ***************** //
+        const VERSION_BYTE_SEED: u8 = 18 << 3;
+        let mut bytes_private = vec![VERSION_BYTE_SEED];
+        // Combine the byte version and the ED25519 raw private key bytes array
+        &bytes_private.extend_from_slice(&keypair.secret.to_bytes());
+        // Calculate checksum
+        let check_sum_private = State::<XMODEM>::calculate(&bytes_private);
+        // Create a buffer to combine byte version : ED25519 raw key : checksum
+        let mut bytes_buffer_private = BytesMut::with_capacity(1024);
+        bytes_buffer_private.put(&bytes_private);
+        bytes_buffer_private.put_u16_le(check_sum_private);
+        // Base 32 encode the private key
+        let private_key = encode(RFC4648 { padding: false }, &bytes_buffer_private);
+
+        (public_key, private_key)
+    }
 }
 
-/// Stellar vanity wallet generator.
-///
```

Comment:
While this does not 100% make sense to be released as a crate and hence a library like I originally did and still do, I believe we should update these docs so that they update with the rust docs. Maybe not in this PR, but in a later PR.

Reply:
Yes, of course I'll update the docs, I just wanted to open the PR as fast, as I can to reach agreement on the implementation first 👍

Reply:
Awesome, merged then

Reply:
See #11